MinVer/build/MinVer.targets

```diff
@@ -11,7 +11,8 @@
     <NoWarn>$(NoWarn);NU5105</NoWarn>
   </PropertyGroup>
 
-  <Target Name="MinVer" BeforeTargets="BeforeCompile;GetAssemblyVersion;CoreCompile;GenerateNuspec" Condition="'$(DesignTimeBuild)' != 'true' AND '$(MinVerSkip)' != 'true'">
+  <!-- Use _GetOutputItemsFromPack instead of GenerateNuSpec to make sure that MinVer will run before calculating NuGet outputs -->
+  <Target Name="MinVer" BeforeTargets="BeforeCompile;GetAssemblyVersion;CoreCompile;_GetOutputItemsFromPack" Condition="'$(DesignTimeBuild)' != 'true' AND '$(MinVerSkip)' != 'true'">

```

Comment:
@xoofx I had a look through the structured logs and I believe I understand the problem now. Indeed, the first instances of the incorrect `...1.0.0.nupkg` in the examples you give in https://github.com/adamralph/minver/issues/675 are indeed in the `_GetOutputItemsFromPack` task.

Taking a closer look, I noticed that `_GetOutputItemsFromPack` is a dependency of `GenerateNuspec`, which also has `$(GenerateNuspecDependsOn)` as its _first_ dependency, so if we inject the `MinVer` target into that variable, I think we should be covered.

I also discovered that `_GetOutputItemsFromPack` is run when `Clean` is run, since `_CleanPackageFiles` runs after `Clean` and depends on `_GetOutputItemsFromPack`. At the moment, the packages never get cleaned when `dotnet clean` is called, because MinVer is never executed as part of the target graph, so the versions of the packages is not determined correctly.

I believe https://github.com/adamralph/minver/pull/681 fixes both problems.

What do you think?

Reply:
@xoofx I just remembered about your [previous comment](https://github.com/adamralph/minver/pull/676/files#issuecomment-1049502373):

> > Another fix would be to insert MinVer into GenerateNuspecDependsOn. I just saw that it was done on GetPackageVersionDependsOn but it looks like it could be also added to GenerateNuspecDepends?
> 
> Hm, actually nope, that won't work. `_GetOutputItemsFromPack` is not part of `GenerateNuspecDepends` so it could still make the version calculation wrong.

I think it's fine to use `$(GenerateNuspecDependsOn)` because `GenerateNuspec` looks like this:

```xml
  <Target Name="GenerateNuspec"
          Condition="'$(IsPackable)' == 'true'"
          Inputs="@(NuGetPackInput)" Outputs="@(NuGetPackOutput)"
          DependsOnTargets="$(GenerateNuspecDependsOn);_CalculateInputsOutputsForPack;_GetProjectReferenceVersions;_InitializeNuspecRepositoryInformationProperties">
```

As you can see it depends firstly on `$(GenerateNuspecDependsOn)`, and secondly on `_CalculateInputsOutputsForPack`.

In turn, `_CalculateInputsOutputsForPack` looks like this:

```xml
  <Target Name="_CalculateInputsOutputsForPack" DependsOnTargets="_GetOutputItemsFromPack">
```

It depends on `_GetOutputItemsFromPack`.

This means that if we inject `MinVer` into `$(GenerateNuspecDependsOn)` then we know `MinVer` will run _before_ `_GetOutputItemsFromPack`.

Reply:
Yep, agreed, #681 should fix the problem. I tested it locally and it works fine!